src/mongo/db/storage/csv_file.cpp

```diff
@@ -170,68 +172,99 @@ Metadata CsvFileInput::getMetadata(const std::vector<std::string>& header) {
     return ret;
 }
 
-template <CsvFieldType T>
-void appendTo(BSONObjBuilder& builder, const std::string& fieldName, const std::string& data);
 
 template <>
-void appendTo<CsvFieldType::kInt32>(BSONObjBuilder& builder,
-                                    const std::string& fieldName,
-                                    const std::string& data) {
-    int converted = stoi(data);
-    builder.append(fieldName, stoi(data));
-    if (data.find('.') != std::string::npos) {
-        LOGV2_WARNING(
-            200000405,
-            "{field}: Lossy conversion of double type to int32 type: {original} To {converted}",
-            "field"_attr = fieldName,
-            "original"_attr = data,
-            "convertedTo"_attr = converted);
+void CsvFileInput::appendTo<CsvFieldType::kInt32>(BSONObjBuilder& builder,
+                                                  const std::string& fieldName,
+                                                  const std::string& data) {
+    int converted;
+    size_t idx;
+    try {
+        converted = stoi(data, &idx);
+        builder.append(fieldName, converted);
+    } catch (std::invalid_argument e) {
```

Comment:
[nit] Let's use const& when catching an exception if there's no strong to do otherwise. Same comment on other places as well.
```suggestion
    } catch (const std::invalid_argument& e) {
```

Reply:
Done